lib/src/screens/session/session_screen.dart

```diff
@@ -193,21 +172,16 @@ class _SessionScreenState extends State<SessionScreen> {
   }
 
   Widget _buildFinished(SessionState session) {
+    // In case of issuance during disclosure, another session is open in a screen lower in the stack.
+    // Ignore clientReturnUrl in this case (issuance) and pop immediately.
     if (session.isIssuanceSession && widget.arguments.hasUnderlyingSession) {
-      // In case of issuance during disclosure, another session is open in a screen lower in the stack.
-      // Ignore clientReturnUrl in this case (issuance) and pop immediately.
       WidgetsBinding.instance?.addPostFrameCallback((_) => Navigator.of(context).pop());
       return _buildLoadingScreen(true);
     }
 
-    if (session.isIssuanceSession &&
-        (session.disclosuresCandidates == null || session.disclosuresCandidates!.isEmpty)) {
-      return const IssuanceSuccessScreen(
-        onDismiss: popToHome,
-      );
-    }
-
-    if (session.continueOnSecondDevice && !(session.clientReturnURL?.isPhoneNumber ?? false)) {
+    if (session.didIssueCredentialFromStore &&
+            (session.disclosuresCandidates == null || session.disclosuresCandidates!.isEmpty) ||
```

Comment:
A comment might be nice why you explicitly require `(session.disclosuresCandidates == null || session.disclosuresCandidates!.isEmpty)`.